src/v/redpanda/admin/server.cc

```diff
@@ -1860,6 +1863,21 @@ admin_server::patch_cluster_config_handler(
                       yaml_name,
                       property.format_raw(yaml_value),
                       validation_err.value().error_message());
+                } else if (auto restricted_err = property.check_restricted(val);
+                           restricted_err.has_value() && should_sanction) {
+                    // TODO(oren): prob better to format the whole message like
+                    // the log?
+                    errors[yaml_name] = restricted_err.value().error_message();
+                    vlog(
+                      adminlog.warn,
+                      "Rejected config due to invalid enterprise license {}: "
+                      "'{}' ({})",
+                      yaml_name,
+                      property.format_raw(yaml_value),
+                      restricted_err.value().error_message());
+                    // Rejected
+                    // TODO(oren): I think we don't need to do anything here,
+                    // since the property is set on the next branch?
```

Comment:
nit: remove todo (agree that I think this is all you need)

Reply:
done